src/vs/code/browser/workbench/workbench.ts

```diff
@@ -16,6 +16,25 @@ import { isEqual } from 'vs/base/common/resources';
 import { isStandalone } from 'vs/base/browser/browser';
 import { localize } from 'vs/nls';
 import { Schemas } from 'vs/base/common/network';
+import product from 'vs/platform/product/common/product';
+
+function doCreateUri(path: string, queryValues: Map<string, string>): URI {
+	let query: string | undefined = undefined;
+
+	if (queryValues) {
```

Comment:
node's `querystring.stringify` might be easier to use here

Reply:
Not sure we can use node functions here as this is client side code.